### PR TITLE
BIG-16297 - Adding some methods to the API Client

### PR DESCRIPTION
**Pull request by bc-joshroe**
BIG-16297 - Adding some methods to the API Client

### CODE REVIEW

**Comment by bc-joshroe** [comment]
https://github.com/bigcommerce/bigcommerce-api-php/issues/172

File: src/Bigcommerce/Api/Client.php [hunk 1]
```diff
@@ -1607,4 +1596,297 @@ public static function deletePage($pageId)
...
+    /**
+     * Create Product Review
+     *
+     * @param int $productId
+     * @param array $object
+     * @return mixed
+     */
+    public static function createProductReview($productId, $object)
+    {
+        return self::createResource('/products/'.$productId.'/reviews', $object);
```
Comment (lord2800) [inline src/Bigcommerce/Api/Client.php]: Spacing (rather, the lack thereof).

Comment (bc-joshroe) [inline src/Bigcommerce/Api/Client.php]: fixed

File: src/Bigcommerce/Api/Client.php [hunk 2]
```diff
@@ -1189,14 +1189,15 @@ public static function getSkus($filter = array())
...
+     * Create a Sku
      *
+     * @param $productId
      * @param $object
      * @return mixed
      */
-    public static function createSku($object)
+    public static function createSku($productId, $object)
     {
-        return self::createResource('/product/skus', $object);
```
Comment (bc-sandeep) [inline src/Bigcommerce/Api/Client.php]: thanks :)

File: src/Bigcommerce/Api/Client.php [hunk 3]
```diff
@@ -1189,14 +1189,15 @@ public static function getSkus($filter = array())
     }
 
     /**
-     * Create sku
+     * Create a Sku
```
Comment (bc-bfenton) [inline src/Bigcommerce/Api/Client.php]: if we're changing this it already it should technically be SKU 🍹 

/pedant

File: src/Bigcommerce/Api/Client.php [hunk 4]
```diff
@@ -1607,4 +1596,297 @@ public static function deletePage($pageId)
...
+     */
+    public static function updateMarketingBanner($bannerID, $object)
+    {
+        return self::updateResource('/banners/' . $bannerID, $object);
+    }
+
+    /**
+     * Add a address to the customer's address book.
+     *
+     * @param $customerID
```
Comment (bc-bfenton) [inline src/Bigcommerce/Api/Client.php]: from here down the docblocks are missing most of their param types

File: src/Bigcommerce/Api/Client.php [hunk 5]
```diff
@@ -1607,4 +1596,297 @@ public static function deletePage($pageId)
...
+    }
+
+    /**
+     * Return the option value object that was created.
+     *
+     * @param $object
+     * @param int $option_id
+     * @return mixed
+     */
+    public static function createOptionValue($option_id, $object)
```
Comment (bc-bfenton) [inline src/Bigcommerce/Api/Client.php]: we're using camel case for the rest of these params, can we keep it consistent here & below?

File: src/Bigcommerce/Api/Client.php (hunk 3 above)
Comment (bc-joshroe) [inline src/Bigcommerce/Api/Client.php]: fixed

File: src/Bigcommerce/Api/Client.php (hunk 4 above)
Comment (bc-joshroe) [inline src/Bigcommerce/Api/Client.php]: fixed

File: src/Bigcommerce/Api/Client.php (hunk 5 above)
Comment (bc-joshroe) [inline src/Bigcommerce/Api/Client.php]: whoops, fixed

**Comment by bc-bfenton** [comment]
👍 pending travis

**Pull request merged**